### PR TITLE
add basic end-to-end connectivity test

### PR DESCRIPTION
**Pull request by cpacejo**
This just configures and launches a basic PH pair and sends pings through to check connectivity.  I'll figure out how to link this up to a GitHub runner in a separate PR…

### CODE REVIEW

File: adapter/ph/test/basic-test.sh [hunk 1]
```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/bash
+set -euo pipefail
+
+PH_BIN=$(realpath "$(dirname $0)/../target/debug/ph-test")
+
+ZPR_USER=$USER
+
+A_HOST_ADDR=10.0.0.1
```
Comment (mkolehmainen) [inline adapter/ph/test/basic-test.sh:8]: I assume that the namespace stuff mens that these addresses will never conflict with actual addressed used on the host, right?

**Review by mkolehmainen — COMMENTED** [review]
Sucks you have to be root, but this looks nice and clean to me.

**Review by mkolehmainen — APPROVED** [review]
approve!

File: adapter/ph/test/basic-test.sh (hunk 1 above)
Comment (cpacejo) [reply adapter/ph/test/basic-test.sh:8]: Correct!

**Comment by cpacejo** [comment]
> Sucks you have to be root, but this looks nice and clean to me.

Yes, unfortunately it's needed to get access to network namespaces..  I'll make a task to try to see if there's a way to give a normal user the necessary privileges (and for the test to detect that and then not try to get root).

Thanks!

**Pull request merged**